### DISCUSSION TITLE
Are SHAP calculations in ExplainerDashboard aggregate? What are the ideal sizes for X_background and X when evaluating neural net models?

### DISCUSSION THREAD

**Discussion by Menamonmon** [oegedijk/explainerdashboard#280] [Q&A]
I have two questions here:

Can`ExplainerDashboard` be used to examine the influence of features on a single prediction or does it only provide accurate explanations at an aggregate level for the model as a whole? If that's possible, how big should the `X` and `X_background` datasets be when passing them to `ClassifierExplainer`. 

How big should the `X_background` be if I want to ensure accuracy of the SHAP values for a large training dataset (millions of rows for training)? 
If SHAP calculations only give valuable insights at an aggregate level, similarly, how big should the X dataset when generating the `ClassifierExplainer`? 

Would really appreciate some examples for clarifications on this? So far, I saw in the examples that use the training data as `X_background` and the testing data as `X` but in my case, I have a very large dataset and want to optimize performance vs accuracy and also want to see if getting explanation for individual predictions are possible. Thank you in advance!

**Comment by oegedijk** [comment]
Hi @Menamonmon, the idea of SHAP values are specific to the example, but can then be aggregated to get global insights into your model. 

In terms of the size of X_background, I think this can be pretty small and still get decent results (tens of examples), but this is probably a question better asked on the shap library github.
